example/spec/testAPI/APISuccessSpec.js

```diff
@@ -0,0 +1,144 @@
+  if (retries === 0) {
+    console.log('Out of retries.');
+    return false;
+  }
+  const existsCheck = await conceptExists(CMRLink);
+  console.log(existsCheck, ' ', outcome);
+  if (existsCheck !== outcome) {
+    sleep.sleep(2);
+    console.log('Retrying...');
+    return waitForExist(CMRLink, outcome, (retries - 1));
+  }
+  else {
+    return true;
+  }
+}
+
+describe('The Cumulus API', () => {
+  let workflowExecution = null;
+  const collection = { name: 'MOD09GQ', version: '006' };
+  const provider = { id: 's3_provider' };
+  const inputPayloadFilename = './spec/testAPI/testAPI.input.payload.json';
+  let inputPayload;
+  let granuleId;
+
+  beforeAll(async () => {
+    console.log('Starting API test');
+    granuleId = randomStringFromRegex(granuleRegex);
+
+    console.log(`granule id: ${granuleId}`);
+
+    const inputPayloadJson = fs.readFileSync(inputPayloadFilename, 'utf8');
+    inputPayload = await setupTestGranuleForAPI(config.bucket, granuleId, inputPayloadJson);
+
+    workflowExecution = await buildAndExecuteWorkflow(
+      config.stackName, config.bucket, taskName, collection, provider, inputPayload
+    );
+  });
+
+  afterAll(async () => {
+    // Remove the granule files added for the test
+    await Promise.all(
+      inputPayload.granules[0].files.map((file) =>
+        s3().deleteObject({
+          Bucket: config.bucket, Key: `${file.path}/${file.name}`
+        }).promise())
+    );
+  });
+  it('completes execution with success status', () => {
+    expect(workflowExecution.status).toEqual('SUCCEEDED');
+  });
+
+  it('makes the granule available through the Cumulus API', async () => {
+    const granule = await apiTestUtils.getGranule({
+      prefix: config.stackName,
+      granuleId: inputPayload.granules[0].granuleId
+    });
+
+    expect(granule.granuleId).toEqual(inputPayload.granules[0].granuleId);
+  });
+
+  describe('reingest a granule', () => {
+    // const granuleId = inputPayload.granules[0].granuleId;
+    it('executes with success status', async () => {
+      const response = await apiTestUtils.reingestGranule({
+        prefix: config.stackName,
+        granuleId
+      });
+      expect(response.status).toEqual('SUCCESS');
+    });
+
+    it('successfully reingest a granule', async () => {
+      const granule = await apiTestUtils.getGranule({
+        prefix: config.stackName,
+        granuleId: inputPayload.granules[0].granuleId
+      });
+      const existsFirstTime = await conceptExists(granule.cmrLink);
+      expect(existsFirstTime).toEqual(true);
+      // Remove the granule from CMR
+      await apiTestUtils.removeFromCMR({
+        prefix: config.stackName,
+        granuleId
+      });
+      // Check that the granule was removed
+      const granuleRemoved = await waitForExist(granule.cmrLink, false, 2);
+      expect(granuleRemoved).toEqual(true);
+
+      // Reingest Granule and check that it was re-added to CMR
+      await apiTestUtils.reingestGranule({
```

Comment:
Do we need to remove a granule from CMR in order to reingest or ingest-in-place a granule?

Reply:
How can we tell the granule reingested/ingested-in-place is a different granule from original one?

Reply:
I do remove the granule from CMR (line 126) and then make sure it has been removed (line 131 and 132) before reingesting. Then I know when I find the granule in CMR again (139 adn 140) the reingest (line 135) has worked.